android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt

```diff
@@ -2293,6 +2294,43 @@ class GeckoEngineSessionTest {
         verify(engineSession, atLeastOnce()).reload()
     }
 
+    @Test
+    fun hasCookieBannerRuleForSession() {
+        val engineSession = GeckoEngineSession(
+            mock(),
+            geckoSessionProvider = geckoSessionProvider,
+        )
+        var onSuccessCalled = false
+        var onErrorCalled = false
+
+        var ruleResult = GeckoResult<Boolean>()
+        whenever(geckoSession.hasCookieBannerRuleForBrowsingContextTree()).thenReturn(ruleResult)
+
+        engineSession.hasCookieBannerRuleForSession(
+            onSuccess = { onSuccessCalled = true },
+            onError = { onErrorCalled = true },
+        )
+
+        ruleResult.complete(true)
+        shadowOf(getMainLooper()).idle()
+
+        assertTrue(onSuccessCalled)
+        assertFalse(onErrorCalled)
+
+        ruleResult = GeckoResult<Boolean>()
+        whenever(geckoSession.hasCookieBannerRuleForBrowsingContextTree()).thenReturn(ruleResult)
+
+        engineSession.hasCookieBannerRuleForSession(
+            onSuccess = { onSuccessCalled = true },
+            onError = { onErrorCalled = true },
+        )
+
+        ruleResult.completeExceptionally(IOException())
+        shadowOf(getMainLooper()).idle()
+
+        assertTrue(onErrorCalled)
+    }
```

Comment:
Nit: Maybe we could break down the test in multiple smaller tests that only cover one test case. This could improve readability for future maintenance. Using [backticks](https://stackoverflow.com/questions/67273691/does-kotlin-allow-the-use-of-backtick-names-for-test-classes) (``), could help to clarify the test's aim, in other [tests files](https://github.com/mozilla-mobile/firefox-android/blob/4a86be5594cfa7bb0ff9a132f6c762b31c5fcc3c/android-components/components/compose/awesomebar/src/test/java/mozilla/components/compose/browser/awesomebar/internal/SuggestionFetcherTest.kt#L29) we try to adopt the pattern [Given When Then](https://martinfowler.com/bliki/GivenWhenThen.html).

Reply:
We already have similar tests in the code base, [for example](https://github.com/mozilla-mobile/firefox-android/blob/055b6b67c690d7f8f1741ad4494d61ac73d91ccd/android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt#L1914-L1980) - that one is even longer.

Reply:
That's true, the test is really old, ideally we would like to improve the code base whenever is possible, to make it more maintainable 🧹 ✨  :)